src/services/socket.tsx

```diff
@@ -25,73 +20,44 @@ type EventName =
 
 type EmitterOptions = {
 	roomId: string
-	message: string | ListenerRes[] | string[]
+	message: string | number | ListenerRes[] | string[]
 	userName: string
+	imageNumber?: number
 	localStorageName?: string
 }
 
 const socket = io()
 
-// NOTE: this can be used with any options, both on server and client.
+// NOTE: this emittercan be used with any options, both on server and client.
 // Emmitters, listeners, & server now use options object pattern.
 export function socketEmitter(eventName: string, options: EmitterOptions) {
-	const timeStamp = Date.now().toString()
-	console.log(
-		'%c>>> services-socketEmitter:\n',
-		'color: #f0c',
-		eventName,
-		'\n',
-		options.localStorageName,
-		options,
-	)
+	const timeStamp = Date.now()
+
 	socket.emit(eventName, {
 		roomId: options.roomId,
 		message: options.message,
 		userName: options.userName,
 		timeStamp: timeStamp,
+		imageNumber: options.imageNumber,
 	})
 	if (options.localStorageName)
 		localStorage.setItem(options.localStorageName, JSON.stringify(options.message))
 }
 
-// TODO: consider deleting useSocketEmitterLocal once certain
-// updated socketEmitter with options (above)
-// works as intended and the same way as this hook.
-// I don't think app will need this useEffect to run any emits.
-export function useSocketEmitterLocal(eventName: string, options: EmitterOptions) {
```

Comment:
praise: you made me proud when you explained why you deleted this code !

Reply:
Thanks!! You are a good mentor, I understood an important concept about useEffects:  they cause re-renders, so minimize them as much as possible. While I knew that on a conceptual level, I hadn't really thought about the impact of 2-3 useEffects in a component!!